src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,77 @@
+        List<String[]> parsedCsv = parseCsv(fromFileName);
+        calculateTotals(parsedCsv);
+
+        saveReport(generateReport(), toFileName);
+    }
+
+    private List<String[]> parseCsv(String fromFileName) {
+        List<String[]> parsedCsv = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line = reader.readLine();
+
+            while (line != null) {
+                parsedCsv.add(line.split(CSV_SEPARATOR));
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error: Can't read file " + fromFileName + ".");
+        }
+
+        return parsedCsv;
+    }
+
+    private void calculateTotals(List<String[]> parsedCsv) {
+        supplyTotal = 0;
+        buyTotal = 0;
+
+        for (String[] parsedLine : parsedCsv) {
+            if (SUPPLY_KEY.equals(parsedLine[INDEX_OF_OPERATION])) {
+                supplyTotal += Integer.parseInt(parsedLine[INDEX_OF_AMOUNT]);
+            } else if (BUY_KEY.equals(parsedLine[INDEX_OF_OPERATION])) {
+                buyTotal += Integer.parseInt(parsedLine[INDEX_OF_AMOUNT]);
+            }
+        }
+    }
+
+    private String generateReport() {
+        StringBuilder reportBuilder = new StringBuilder();
+
+        reportBuilder.append(SUPPLY_KEY + CSV_SEPARATOR + supplyTotal + NEW_LINE);
+        reportBuilder.append(BUY_KEY + CSV_SEPARATOR + buyTotal + NEW_LINE);
+        reportBuilder.append(RESULT_KEY + CSV_SEPARATOR + (supplyTotal - buyTotal));
+
+        return reportBuilder.toString();
+    }
 
+    private void saveReport(String report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (IOException e) {
+            System.out.println("Error: Can't write to file " + toFileName + ".");
```

[Don't ignore exceptions](https://github.com/mate-academy/jv-work-with-file/blob/master/checklist.md#dont-ignore-exceptions)